internal/batches/executor/executor.go

```diff
@@ -423,11 +258,17 @@ func (x *executor) do(ctx context.Context, task *Task) (err error) {
 		return
 	}
 
+	// Build the changeset specs.
```

Comment:
This has been moved down so that everything related to the spec-building is close together.